sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs

```diff
@@ -331,13 +331,25 @@ public async Task DeleteAsync_ConditionsFail()
             }
         }
 
+        // It is possible for the service to have responded that the container was already created after
+        // creating the new FileSystem only to have the directory creation fail. This appears to be due to
+        // a propagation delay from the blob endpoint to the dfs endpoint, since the directory creation happens against
+        // the dfs endpoint.
+        // This method will create the directory and retry if a 404 is returned that indicates the filesystem doesn't exist.
+        private async Task<DataLakeDirectoryClient> CreateDirectoryEnsureFileSystemPropagated(DisposingFileSystem fileSystem, string directoryName)
```

Comment:
Why not just pass the `FileSystem` in?  We're not using the wrapper.

Is it worth just calling this `CreateDirectory`?

Reply:
Good call